precompiles/randomness/Randomness.sol

```diff
@@ -0,0 +1,148 @@
+pragma solidity >=0.8.0;
+
+/// @dev Maximum number of random words being requested
+uint32 constant MAX_RANDOM_WORDS = 100;
+/// @dev Minimum number of blocks before a request can be fulfilled
+uint32 constant MIN_DELAY_BLOCKS = 2;
+/// @dev Maximum number of blocks before a request can be fulfilled
+uint32 constant MAX_DELAY_BLOCKS = 2000;
+
+interface Randomness {
+    /// @notice The status of the request
+    /// @param Pending The request cannot be fulfilled yet
+    /// @param Ready The request is ready to be fulfilled
+    /// @param Expired The request has expired
+    enum RequestStatus {
+        Pending,
+        Ready,
+        Expired
+    }
+
+    /// @notice The type of randomness source
+    /// @param LocalVRF Randomness VRF using the parachain material as seed
+    /// @param RelayBabeEpoch Randomness VRF using relay material from previous epoch
+    enum RandomnessSource {
+        LocalVRF,
+        RelayBabeEpoch
+    }
+
+    /// @notice The request details
+    /// @param id The id of the request
+    /// @param refundAddress The address receiving the left-over fees after the fulfillment
+    /// @param contractAddress The address of the contract being called back during fulfillment
+    /// @param fee The amount to set aside to pay for the fulfillment
+    /// @param gasLimit The gas limit to use for the fulfillment
+    /// @param salt A string being mixed with the randomness seed to obtain different random words
+    /// @param numWords The number of random words requested (from 1 to MAX_RANDOM_WORDS)
+    /// @param randomnessSource The type of randomness source used to generate the random words
+    /// @param fulfillmentBlock The parachain block number at which the request can be fulfilled (for LocalVRF only)
+    /// @param fulfillmentEpochIndex The relay epoch index at which the request can be fulfilled (for RelayBabeEpoch)
+    /// @param expirationBlock The parachain block number at which the request expires
+    /// @param status The current status of the request
+    struct Request {
+        uint256 id;
+        address refundAddress;
+        address contractAddress;
+        uint256 fee;
+        uint256 gasLimit;
+        bytes32 salt;
+        uint32 numWords;
+        RandomnessSource randomnessSource;
+        uint32 fulfillmentBlock;
+        uint32 fulfillmentEpochIndex;
+        uint32 expirationBlock;
+        RequestStatus status;
+    }
+
+    /// Return the current relay epoch index
+    /// @dev An epoch represents real time and not a block number
+    /// @dev Currently, time between epoch changes cannot be longer than:
+    /// @dev  - Kusama/Westend/Rococo: 600 relay blocks (1 hour)
+    /// @dev  - Polkadot: 2400 relay blocks (4 hours)
+    /// Selector: 81797566
+    function relayEpochIndex() external view returns (uint64);
+
+    /// @param requestId The id of the request to check
+    /// @return request The request
+    /// Selector: c58343ef
+    function getRequest(uint256 requestId)
```

Comment:
This selector is not implemented

Reply:
I believe @4meta5 is still working on implementing those

Reply:
I think you should write the enum or const for those value (also you need to merge the crystalin-vrf-lottery branch to get the exact enum expected